patches/minecraft/net/minecraft/item/ItemStack.java.patch

```diff
@@ -26,26 +26,37 @@
  
 -    public ItemStack(Item p_i1881_1_, int p_i1881_2_, int p_i1881_3_)
 +    public ItemStack(Item p_i1881_1_, int p_i1881_2_, int p_i1881_3_){ this(p_i1881_1_, p_i1881_2_, p_i1881_3_, null); }
-+    public ItemStack(Item p_i1881_1_, int p_i1881_2_, int p_i1881_3_, NBTTagCompound capNBT)
++    public ItemStack(Item p_i1881_1_, int p_i1881_2_, int p_i1881_3_, @Nullable NBTTagCompound capNBT)
      {
 +        this.capNBT = capNBT;
          this.field_151002_e = p_i1881_1_;
          this.field_77991_e = p_i1881_3_;
          this.field_77994_a = p_i1881_2_;
-@@ -105,10 +111,12 @@
-     private void func_190923_F()
-     {
-         this.field_190928_g = this.func_190926_b();
-+        this.delegate = this.field_151002_e != null ? this.field_151002_e.delegate : null;
+@@ -100,6 +106,7 @@
+         }
+ 
+         this.func_190923_F();
++        this.forgeInit(p_i1881_1_);
```

Comment:
Also if I call `this.item`, it routes to `this.getItem()` which triggers `Block`'s static initializer which crashes. I assume @cpw is doing some scary thing to make that happen. So `forgeInit` takes item as a parameter to avoid calling `this.item`.